core/src/HIP/Kokkos_HIP_Vectorization.hpp

```diff
@@ -76,17 +76,19 @@ struct in_place_shfl_op {
     union conv_type {
       Scalar orig;
       shfl_type conv;
+      // This should be fine, members get explicitly reset, which changes the
+      // active member
+      KOKKOS_FUNCTION conv_type() { conv = 0; }
```

Comment:
So here just copying the Cuda implementation
https://github.com/kokkos/kokkos/blob/20a090bf6a7268f6d46521b686e46da3b40e925a/core/src/Cuda/Kokkos_Cuda_Vectorization.hpp#L79-L102

It was updated in #2991 
I suppose you don't want to keep the FIXME L68?